CharlieBackend.Business/Services/StudentService.cs

```diff
@@ -170,6 +170,10 @@ public async Task<Result<StudentDto>> GetStudentByAccountIdAsync(long accountId)
 
         public async Task<Result<StudentDto>> GetStudentByIdAsync(long studentId)
         {
+            var allStudents = await GetAllStudentsAsync();
+            var maxId = allStudents.Data.Select(st => st.Id).Max();
+            if (studentId > maxId)
+                return Result<StudentDto>.GetError(ErrorCode.NotFound, $"This id = {studentId} is bigger than maximum student id = {maxId}");
```

Comment:
It doesn't matter if `studentId` is in range or not. `StudentRepository` uses `FirstOrDefault()` method which returns `null` when student with the given Id is not found:

![check](https://user-images.githubusercontent.com/38729881/105612572-bca80300-5dc5-11eb-9e0d-9bbf531e6b41.png)

So, basically, all we need to do here is add check if student is found or not (if `student` is `null`)
@filinskyi

Reply:
> It doesn't matter if `studentId` is in range or not. `StudentRepository.GetByIdAsync()` uses `FirstOrDefault()` method which returns `null` when student with the given Id is not found:

Yes, that is right.

> So, basically, all we need to do here is add check if student is found or not (if `student` is `null`)
I think it's better to ask @Hlib-mats  what is the actual problem.